### DISCUSSION TITLE
Repeated commands move window to same position on next screen

### DISCUSSION THREAD

**Discussion by kitschpatrol** [rxhanson/Rectangle#374] [Ideas]
Hi, thanks for a great app!

I realize there's already some complexity around the behavior of repeated commands, and you're not looking for any new features, but I wanted to share a quick implementation of a feature I liked from Divvy where repeated commands moved the the window to the _same_ position on the next display.

This is somewhat similar to `subsequentExecutionMode` `1` and `3`, but instead of wrapping spatially to the next display (e.g. from right edge to left edge), the window teleports to the same normalized location on the next display. This opens up additional commands for screen cycling beyond just left and right. (E.g. a second "maximize" command cycles between displays.)

This is also somewhat similar to the "Next Display" and "Previous Display" commands, but the approach here preserves (normalized) window size and position, and lets you overload existing shortcuts.

It's implemented as `subsequentExecutionMode 4` in [this branch of my fork](https://github.com/kitschpatrol/Rectangle/tree/feature/cycle-displays), and I've taken a stab at further explanation of the various `subsequentExecutionMode` behaviors in [this section of the readme](https://github.com/kitschpatrol/Rectangle/tree/feature/cycle-displays#repeated-command-behavior).

I'm happy to test / refine this further if there's interest in a PR. (The current implementation involves some abuse of `recordAction`...)

**Comment by rxhanson** [comment]
Thanks for looking to contribute! This makes sense to me, and I think it would make a good addition to the app. Feel free to make PR and I will give things a run through once you have it up.

**Reply by kitschpatrol** [reply]
Thanks — and pardon the delay. Opened PR https://github.com/rxhanson/Rectangle/pull/394.
